useragent/cicd.go

```diff
@@ -0,0 +1,57 @@
+package useragent
+
+import "os"
+
+type envVar struct {
+	name          string
+	expectedValue string
+}
+
+type cicdProvider struct {
+	// The name of the CI/CD provider. This is the name included in the user
+	// agent string.
+	name string
+
+	// The env vars that are expected to be set in the CI/CD provider's runner.
+	envVars []envVar
+}
+
+// cicdProviders is a list of CI/CD providers and their env vars we can rely on
+// to detect them.
+var cicdProviders = []cicdProvider{
+	{"github", []envVar{{"GITHUB_ACTIONS", "true"}}},
+	{"gitlab", []envVar{{"GITLAB_CI", "true"}}},
+	{"jenkins", []envVar{{"JENKINS_URL", ""}}},
+	{"azure-devops", []envVar{{"TF_BUILD", "True"}}},
+	{"circle", []envVar{{"CIRCLECI", "true"}}},
+	{"travis", []envVar{{"TRAVIS", "true"}}},
+	{"bitbucket", []envVar{{"BITBUCKET_BUILD_NUMBER", ""}}},
+	{"google-cloud-build", []envVar{{"PROJECT_ID", ""}, {"BUILD_ID", ""}, {"PROJECT_NUMBER", ""}, {"LOCATION", ""}}},
+	{"aws-code-build", []envVar{{"CODEBUILD_BUILD_ARN", ""}}},
+	{"tf-cloud", []envVar{{"TFC_RUN_ID", ""}}},
+}
```

Comment:
We should have a method that returns the mutable variable since it's easy for some other method (down the line) to change the values.

I think we should abstract out the list to another file 
Have a method that returns this list instead of modifying this mutable variable.

Reply:
done